### PR TITLE
grab is driven by timer rather than update, load tablet model with high priority

### PR DESCRIPTION
**Pull request by sethalves**
- adjust handControllerGrab so it works better while content is loading
- load tablet model at a higher priority than usual.

### CODE REVIEW

File: scripts/system/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -4032,12 +4034,12 @@ function updateWrapper(){
         updateTotalWork = 0;
     }
 
+    Script.setTimeout(updateWrapper, UPDATE_SLEEP_MS);
 }
 
-Script.update.connect(updateWrapper);
+Script.setTimeout(updateWrapper, UPDATE_SLEEP_MS);
```
Comment (hyperlogic) [inline scripts/system/controllers/handControllerGrab.js]: Shouldn't this be setInterval?

**Comment by hifi-gustavo** [comment]
Build console output for commit: ec687bdb4b90ea6ec673dc212581618b3e9a7cf6 available at: https://gist.github.com/35f06ae7ce88efa0699e1debe3256402<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-ec687bd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-ec687bd.dmg

File: scripts/system/controllers/handControllerGrab.js (hunk 1 above)
Comment (ZappoMan) [inline scripts/system/controllers/handControllerGrab.js]: maybe he only wants one callback. ;)

Comment (hyperlogic) [inline scripts/system/controllers/handControllerGrab.js]: Maybe I'm missing something, I can't imagine this working at all.

Comment (sethalves) [inline scripts/system/controllers/handControllerGrab.js]: no, setInterval can build up when things are bogging down.  setTimeout is better, here.

Comment (sethalves) [inline scripts/system/controllers/handControllerGrab.js]: http://thecodeship.com/web-development/alternative-to-javascript-evil-setinterval/

Comment (hyperlogic) [inline scripts/system/controllers/handControllerGrab.js]: setTimeout is only calls the update function, once, tho

Comment (hyperlogic) [inline scripts/system/controllers/handControllerGrab.js]: Nevermind, I missed the following line:

     Script.setTimeout(updateWrapper, UPDATE_SLEEP_MS);

Comment (sethalves) [inline scripts/system/controllers/handControllerGrab.js]: the end of the update function also does a setTimeout.  It works!  I've run it!

Comment (sethalves) [inline scripts/system/controllers/handControllerGrab.js]: it's sort of like looping via tail-recursion

**Comment by hyperlogic** [comment]
why is this marked as DO NOT MERGE?

**Comment by sethalves** [comment]
because I don't know if it's going to help fix the problem it's intended to fix.  It would be great if ryan could run it and do what he was doing yesterday (changing domains, open tablet before loading is finished, etc).  If it doesn't help anything, there's not a good reason to merge it.

**Comment by RyanDowne** [comment]
testing

**Comment by RyanDowne** [comment]
This doesn't seem to help. When I change domains and try to open my tablet, sometimes it just doesn't show up and when it does it can't be grabbed nor touched. I can show you on this build if you wanna watch on my desktop.

**Comment by hifi-gustavo** [comment]
Build console output for commit: aeaa95b7a09d349c20b016343da4cd15b03fb017 available at: https://gist.github.com/35f06ae7ce88efa0699e1debe3256402<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-aeaa95b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-aeaa95b.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 93f9716557736faafe142caf31c583a85228676e available at: https://gist.github.com/35f06ae7ce88efa0699e1debe3256402<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-93f9716.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-93f9716.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 467505e4217784be0d74f8457b7be200297971b1 available at: https://gist.github.com/35f06ae7ce88efa0699e1debe3256402<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-467505e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-467505e.dmg

**Comment by anshuman64** [comment]
QA Pass in my book. @RyanDowne you want to give it a shot?

**Comment by anshuman64** [comment]
Asking @sharonsein to look at this as well

**Comment by RyanDowne** [comment]
I tested this again and it made a pretty big difference. There were still situations where when loading content in a new domain and summoning the tablet, it appears then goes away but I can't get it to happen all the time. I also saw situations where the web-surface arrives before the model. That being said, the responsiveness of the tablet when loading in this PR is a big improvement and goes a long way to making our product feel less broken to end users. I did not do a thorough QA pass.

**Comment by anshuman64** [comment]
Comment from QA Tester sharonsein:
"Can confirm that it takes a while for the stylus to load when waiting for a domain to load. Trying to select things on the tablet before they appear makes it incredibly easy to select a whole bunch of things the user doesn't intend, which can lead to causing HF to crash. Scrolling through the go-to menu without them is especially frustrating."

We will do another test to see how this change affects frame rate before merging. After merging, we will open another ticket to improve load time of styluses (for lower end machines).

**Comment by anshuman64** [comment]
To test
0. Get the latest version of the  PR
1. Go to dev-playa domain
2. Go to Settings and enable "Advanced Menus" and "Developer Menus"
3. Go to the Edit Menu (not the tablet app) and click "Reload Content"
4. As things are loading: 
a) Try using the tablet--it should function well
b) Try doing other stuff to see if everything behaves as it should be
c) Compare the effects of this PR on the frame rate while things are loading

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/4265/Can-t-grab-tablet-Buttons-unresponsive

**Comment by Nex-Pro** [comment]
Testing this. Win 10 and Vive.

**Comment by Nex-Pro** [comment]
@sethalves can we get this up to date with current protocols? This is going to need some teleporting between domains.

**Comment by hifi-gustavo** [comment]
Build console output for commit: dc569051f9d79c41b63182c8944c5df0bb219fa8 available at: https://gist.github.com/35f06ae7ce88efa0699e1debe3256402<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-dc56905.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10257-dc56905.dmg

**Comment by anshuman64** [comment]
Just using Stats, no noticeable depreciation in frame rate or render rate. Calling this a QA Pass.

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/4265/Can-t-grab-tablet-Buttons-unresponsive

**Pull request merged**